### PR TITLE
Feat: Add OpenAI and Azure voice services

### PR DESCRIPTION
**Pull request by pyarmak**
I have taken the liberty of implementing two additional AI voice services, namely, OpenAI (which you basically had already finished) and Azure.
It's been tested to work but it's a quick and dirty addition so may need some polish.

### CODE REVIEW

File: src/ServiceConfigurationModal.ts [hunk 1]
```diff
@@ -40,19 +47,88 @@ export class ServiceConfigurationModal extends Modal {
...
+				.addText(async text => {
+					text
+						.setValue(this.plugin.settings.services.azure.key)
+						.onChange(async value => {
+							this.plugin.settings.services.azure.key = value;
+							await this.plugin.saveSettings();
+					});
+				});
+			new Setting(contentEl)
+				.setName('Speech Region')
```
Comment (joethei) [inline src/ServiceConfigurationModal.ts:66]: Use sentence case

File: src/ServiceConfigurationModal.ts [hunk 2]
```diff
@@ -40,19 +47,88 @@ export class ServiceConfigurationModal extends Modal {
...
 							await this.plugin.saveSettings();
 					});
-				})
-			;
+				});
+		}
+
+		if (service === 'azure') {
+			new Setting(contentEl)
+				.setName('API Key')
```
Comment (joethei) [inline src/ServiceConfigurationModal.ts:55]: Use sentence case

File: src/settings.ts [hunk 3]
```diff
@@ -122,7 +148,48 @@ export class TTSSettingsTab extends PluginSettingTab {
 					.onClick(() => {
 						new ServiceConfigurationModal(this.plugin).open();
 					});
-			});*/
+			});
+		
+		const servicesEl = containerEl.createDiv("settings-banner", (banner) => {
```
Comment (joethei) [inline src/settings.ts:153]: Use `new Setting(containerEl).setName('Configured services').setHeading();` instead.

**Review by joethei — CHANGES_REQUESTED** [review]

**Comment by pyarmak** [comment]
I'm actually working on a couple of more features. I went down a rabbit hole.
I will update my code with your suggestions and will update the PR once I have completed the other features.
Namely, I'm adding a custom regex filter for the input text (I'll add it under advanced features in the settings) and the ability to play/pause/stop/seek the generated audio output. To that end, I am modifying the createMenu function and completely removing the menu and instead dynamically adding buttons and the seek bar to the status bar.

**Comment by joethei** [comment]
Could you put those features in a separate PR?

**Pull request closed without merging**